server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -3631,6 +3658,40 @@ void sc_osc_handler::do_asynchronous_command(World* world, void* replyAddr, cons
                                            shared_endpoint);
 }
 
+// called from RT thread, perform in NRT thread, free in RT thread
+template <bool realtime> void handle_message_from_RT(FifoMsg& msg) {
+    // see handle_asynchronous_command()
+    spin_lock::scoped_lock lock(system_callback_allocator_lock);
+
+    cmd_dispatcher<realtime>::fire_system_callback([msg]() mutable {
+        msg.Perform();
+
+        cmd_dispatcher<realtime>::fire_rt_callback([msg]() mutable { msg.Free(); });
+    });
+}
+
+void sc_osc_handler::send_message_from_RT(World* world, FifoMsg& msg) const {
+    if (world->mRealTime)
+        handle_message_from_RT<true>(msg);
+    else
+        handle_message_from_RT<false>(msg);
+}
+
+// called from NRT thread, perform in RT thread, free in NRT thread
+template <bool realtime> void handle_message_to_RT(FifoMsg& msg) {
+    cmd_dispatcher<realtime>::fire_rt_callback([msg]() mutable {
+        msg.Perform();
+
+        cmd_dispatcher<realtime>::fire_system_callback([msg]() mutable { msg.Free(); });
+    });
+}
+
+void sc_osc_handler::send_message_to_RT(World* world, FifoMsg& msg) const {
```

Comment:
`world` in `send_message_to_RT` and `send_message_from_RT` (both overloads) should be `const World*`